src/main/java/net/rptools/maptool/client/ui/AbstractTokenPopupMenu.java

```diff
@@ -143,17 +143,46 @@ protected JMenu createLightSourceMenu() {
       }
       menu.addSeparator();
     }
+
+    // Add unique light sources for the token.
+    {
+      JMenu subMenu = new JMenu("Unique");
+
+      List<LightSource> lightSources = new ArrayList<>(tokenUnderMouse.getUniqueLightSources());
+      Collections.sort(lightSources);
+
+      LIGHTSOURCES:
```

Comment:
This may be just my personal opinion but continue label looks bad for me. (To much like goto.) How about you add an LightSource.IsGM() and use that here?

Reply:
Yeah, I should rewrite these loops. The menu wasn't a focus for me, so I didn't want to change the behaviour, but these are just not correct as they are.

E.g., consider a partially GM-only aura: `Test: aura circle 20#ff0000 GM 40#00ff00`. This shouldn't be excluded from the menu.